### PR TITLE
Collector frontend

### LINKED ISSUE #508: Support ODD Collector
We need to support ODD collector Data Source's registration support

### PR DESCRIPTION
**Pull request by alinamiryuk**
- [ ] **Breaking change?** (if so, please describe the impact and migration path for existing applications:)

**What changes did you make?** (Give an overview)
- CollectorsList, CollectorForm, CollectorItem, CollectorSkeletonItem
- interfaces, actions, thunks, reducer were created for CollectorsList component

**Is there anything you'd like reviewers to focus on?**
Resolved #508 

**How Has This Been Tested?** (put an "X" next to an item)

- [X] Manually(please, describe, when necessary)

**Checklist** (put an "X" next to an item, otherwise PR will fail)
- [X] I have performed a self-review of my own code

Check out [Contributing](https://github.com/opendatadiscovery/odd-platform/blob/main/CONTRIBUTING.md) and [Code of Conduct](https://github.com/opendatadiscovery/odd-platform/blob/main/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: odd-platform-ui/src/components/Management/CollectorsList/CollectorForm/CollectorForm.tsx [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+  collector?: Collector;
+  registerCollector: (
+    params: CollectorApiRegisterCollectorRequest
+  ) => Promise<Collector>;
+  updateCollector: (
+    params: CollectorApiUpdateCollectorRequest
+  ) => Promise<Collector>;
+}
+
+export type CollectorFormDataValues = CollectorFormData;
```
Comment (Leshe4ka) [inline odd-platform-ui/src/components/Management/CollectorsList/CollectorForm/CollectorForm.tsx:29]: I think overriding this type is useless

File: odd-platform-ui/src/components/Management/CollectorsList/CollectorForm/CollectorForm.tsx [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+          error: response.statusText || 'Unable to register datasource',
+        });
+      }
+    );
+  };
+
+  const formTitle = (
+    <Typography variant="h4" component="span">
+      {collector ? 'Edit ' : 'Add '}
+      Datasource
```
Comment (Leshe4ka) [inline odd-platform-ui/src/components/Management/CollectorsList/CollectorForm/CollectorForm.tsx:99]: Please use the naming associated with the collector

File: odd-platform-ui/src/components/Management/CollectorsList/CollectorForm/CollectorForm.tsx [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+      />
+      <Controller
+        name="description"
+        control={control}
+        render={({ field }) => (
+          <AppTextField
+            {...field}
+            sx={{ mt: 1.25 }}
+            label="Description"
+            placeholder="Datasource description"
```
Comment (Leshe4ka) [inline odd-platform-ui/src/components/Management/CollectorsList/CollectorForm/CollectorForm.tsx:148]: Please use the naming associated with the collector

File: odd-platform-ui/src/components/Management/CollectorsList/CollectorForm/CollectorForm.tsx [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+        )}
+      />
+    </form>
+  );
+
+  const formActionButtons = () => (
+    <AppButton
+      size="large"
+      type="submit"
+      form="datasource-create-form"
```
Comment (Leshe4ka) [inline odd-platform-ui/src/components/Management/CollectorsList/CollectorForm/CollectorForm.tsx:167]: Please use the naming associated with the collector

File: odd-platform-ui/src/components/Management/CollectorsList/CollectorItem/CollectorItem.tsx [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+                color="primaryLight"
+                startIcon={<EditIcon />}
+                sx={{ mr: 1 }}
+              >
+                Edit
+              </AppButton>
+            }
+          />
+          <ConfirmationDialog
+            actionTitle="Are you sure you want to delete this datasource?"
```
Comment (Leshe4ka) [inline odd-platform-ui/src/components/Management/CollectorsList/CollectorItem/CollectorItem.tsx:56]: Please use the naming associated with the collector

File: odd-platform-ui/src/components/Management/CollectorsList/CollectorItem/CollectorItem.tsx [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+            actionName="Delete"
+            actionText={
+              <Typography variant="subtitle1">
+                Delete &quot;{collector.name}&quot; datasource?
```
Comment (Leshe4ka) [inline odd-platform-ui/src/components/Management/CollectorsList/CollectorItem/CollectorItem.tsx:60]: Please use the naming associated with the collector

File: odd-platform-ui/src/components/Management/CollectorsList/CollectorItem/CollectorItemToken/CollectorItemToken.tsx [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+      }),
+    [collector]
+  );
+
+  return (
+    <TokenContainer>
+      <Token $isHidden={isHidden}>{collector.token.value}</Token>
+      {isHidden ? (
+        <ConfirmationDialog
+          actionTitle="Are you sure you want to regenerate token for this datasource?"
```
Comment (Leshe4ka) [inline odd-platform-ui/src/components/Management/CollectorsList/CollectorItem/CollectorItemToken/CollectorItemToken.tsx:42]: Please use the naming associated with the collector

File: odd-platform-ui/src/redux/thunks/collectors.thunks.ts [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+  (
+    response: CollectorList,
+    request: CollectorApiGetCollectorsListRequest
+  ) => ({
+    ...response,
+    pageInfo: {
+      ...response.pageInfo,
+      page: request.page,
+      hasNext: request.size * request.page < response.pageInfo.total,
+    },
```
Comment (Leshe4ka) [inline odd-platform-ui/src/redux/thunks/collectors.thunks.ts:38]: As far as I know, now pagination comes from the backend correctly. I think now it is not necessary to consider hasNext in this way.

CC: @damirabdul

**Review by Leshe4ka — APPROVED** [review]

File: odd-platform-ui/src/components/Management/CollectorsList/CollectorForm/CollectorForm.tsx (hunk 1 above)
Comment (alinamiryuk) [reply odd-platform-ui/src/components/Management/CollectorsList/CollectorForm/CollectorForm.tsx:29]: done

File: odd-platform-ui/src/components/Management/CollectorsList/CollectorForm/CollectorForm.tsx (hunk 4 above)
Comment (alinamiryuk) [reply odd-platform-ui/src/components/Management/CollectorsList/CollectorForm/CollectorForm.tsx:167]: done

File: odd-platform-ui/src/components/Management/CollectorsList/CollectorForm/CollectorForm.tsx (hunk 3 above)
Comment (alinamiryuk) [reply odd-platform-ui/src/components/Management/CollectorsList/CollectorForm/CollectorForm.tsx:148]: done

File: odd-platform-ui/src/components/Management/CollectorsList/CollectorItem/CollectorItem.tsx (hunk 6 above)
Comment (alinamiryuk) [reply odd-platform-ui/src/components/Management/CollectorsList/CollectorItem/CollectorItem.tsx:60]: done

File: odd-platform-ui/src/components/Management/CollectorsList/CollectorItem/CollectorItemToken/CollectorItemToken.tsx (hunk 7 above)
Comment (alinamiryuk) [reply odd-platform-ui/src/components/Management/CollectorsList/CollectorItem/CollectorItemToken/CollectorItemToken.tsx:42]: done

File: odd-platform-ui/src/components/Management/CollectorsList/CollectorItem/CollectorItem.tsx [hunk 9]
```diff
@@ -0,0 +1,117 @@
...
+} from 'generated-sources';
+import LabeledInfoItem from 'components/shared/LabeledInfoItem/LabeledInfoItem';
+import ConfirmationDialog from 'components/shared/ConfirmationDialog/ConfirmationDialog';
+import EditIcon from 'components/shared/Icons/EditIcon';
+import DeleteIcon from 'components/shared/Icons/DeleteIcon';
+import AppButton from 'components/shared/AppButton/AppButton';
+import AppTooltip from 'components/shared/AppTooltip/AppTooltip';
+import CollectorFormDialogContainer from 'components/Management/CollectorsList/CollectorForm/CollectorFormContainer';
+import CollectorItemTokenContainer from './CollectorItemToken/CollectorItemTokenContainer';
+import * as S from './CollectorItemStyles';
```
[the remark this answers is not in the document]
Comment (alinamiryuk) [reply odd-platform-ui/src/components/Management/CollectorsList/CollectorItem/CollectorItem.tsx:15]: done

File: odd-platform-ui/src/redux/thunks/collectors.thunks.ts (hunk 8 above)
Comment (alinamiryuk) [reply odd-platform-ui/src/redux/thunks/collectors.thunks.ts:38]: done

**Pull request merged**